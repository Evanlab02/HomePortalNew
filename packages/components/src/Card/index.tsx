import { Card, ConfigProvider, ThemeConfig } from "antd";
import { PropsWithChildren, ReactNode } from "react";

const theme: ThemeConfig = {
	token: {
		colorPrimary: "#1677ff",
		colorText: "#fff"
	},
	components: {
		Card: {
			colorBgContainer: "#001529",
			colorBorderSecondary: "rgba(255, 255, 255, 0.65)"
		}
	}
};

export interface CardProps extends PropsWithChildren {
    title: string;
    extra?: ReactNode;
}

export default function HpCard(props: Readonly<CardProps>) {
	const { title, extra, children } = props;

	return (
		<ConfigProvider theme={theme}>
			<Card
				title={title}
				extra={extra}
			>
				{children}
			</Card>
		</ConfigProvider>
	);
}