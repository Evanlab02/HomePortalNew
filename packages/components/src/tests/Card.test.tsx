import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import Card from "../Card";

test("Card renders correctly", async () => {
	const { findByTestId } = render(
		<div data-testid="testing-hp-card">
			<Card
				title="Hello World"
				extra={<a href="/link/to/somewhere">Random Link</a>}
			>
				<p>This is the card description, and goodbye world!</p>
			</Card>
		</div>
	);

	const card = await findByTestId("testing-hp-card");
	expect(card).toMatchSnapshot();
});