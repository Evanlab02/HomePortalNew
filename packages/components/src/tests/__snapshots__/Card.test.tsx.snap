// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card renders correctly 1`] = `
<div
  data-testid="testing-hp-card"
>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-w5lxuz"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          Hello World
        </div>
        <div
          class="ant-card-extra"
        >
          <a
            href="/link/to/somewhere"
          >
            Random Link
          </a>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <p>
        This is the card description, and goodbye world!
      </p>
    </div>
  </div>
</div>
`;
